tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalDataInfo.java

```diff
@@ -344,11 +346,23 @@ public BasicLanguageData setType(Type type) {
             return this;
         }
 
-        public BasicLanguageData setScripts(String scriptTokens) {
-            return setScripts(
-                    scriptTokens == null
-                            ? null
-                            : Arrays.asList(WHITESPACE_PATTERN.split(scriptTokens)));
+        // Adding scripts but leaving 0 as a placeholder when there is no population data
+        // input: a whitespace-separated list of scripts
+        public BasicLanguageData setScriptsWithoutPopulation(String scriptTokens) {
+            List<String> scripts = new ArrayList<>();
+            if (scriptTokens != null) {
+                scripts = Arrays.asList(WHITESPACE_PATTERN.split(scriptTokens));
+            }
+            return setScriptsWithoutPopulation(scripts);
+        }
+
+        // Adding scripts but leaving 0 as a placeholder when there is no population data
+        public BasicLanguageData setScriptsWithoutPopulation(Collection<String> scripts) {
+            Map<String, Integer> scriptsByPopulation = new TreeMap<>();
+            for (String script : scripts) {
+                scriptsByPopulation.put(script, 0);
+            }
+            return setScripts(scriptsByPopulation);
```

Comment:
Better to build once, make immutable.

Reply:
I know generally what you mean but I'm not sure how you want me to do this. Do you want me to avoid calling setScripts and include the code here as well? Personally I prefer DRY code over WET.